src/main/java/com/snowflake/kafka/connector/internal/FileNameUtils.java

```diff
@@ -192,4 +196,39 @@ private static String readFromFileName(String fileName, int index) {
 
     return matcher.group(index);
   }
+
+  /**
+   * Find gaps in offset ranges.
+   *
+   * @param filenames list of files
+   * @return continuous and missing offsets for given filenames
+   */
+  static OffsetContinuityRanges searchForMissingOffsets(List<String> filenames) {
+    List<Pair<Long, Long>> missingOffsets = new ArrayList<>();
+
+    List<Pair<Long, Long>> continuousOffsets =
+        filenames.stream()
+            .map(
+                file ->
+                    Pair.of(
+                        FileNameUtils.fileNameToStartOffset(file),
+                        FileNameUtils.fileNameToEndOffset(file)))
+            .collect(Collectors.toList());
+
+    for (int i = 0; i < continuousOffsets.size(); i++) {
+      Pair<Long, Long> current = continuousOffsets.get(i);
+
+      // The first range is skipped
+      if (i == 0) {
```

Comment:
then why don't we just start from i = 1 ? :D

Reply:
Before the last refactor it made sense xD Sure, will fix.